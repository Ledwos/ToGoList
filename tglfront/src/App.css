
#title {
    margin: 5px 0px 0px 10px;
    font-size: 30px;
}

nav {
    display: inline-flex;
    justify-content: space-between;
    /* border: 2px solid red; */
    width: 100%;
    background-color: aqua;

}

.navMain {
    display: inline-flex;
    margin-right: 5%;
    /* border: 2px solid orange; */
}

.navItem {
    display: inline-flex;
    list-style-type: none;
    /* border: 2px solid blue; */
    padding: 2px 5px;
    margin: 0px 5px;
}

.navItem:hover {
    cursor: pointer;
}