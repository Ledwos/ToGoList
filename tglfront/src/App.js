import React, { useState, useEffect } from 'react';
import './App.css';
import TaskComp from './components/TaskComp';
import LogInComp from './components/LogInComp';


const App = () => {
  const [loggedIn, setloggedIn] = useState(true);
  const [uId, setuId] = useState(5);

  if (loggedIn) {
    return <TaskComp user={uId}/>
  } else {
    return <LogInComp />
  }
}

export default App;
