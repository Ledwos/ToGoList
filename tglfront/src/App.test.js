import React from 'react';
// import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import App from './App';
import renderer from 'react-test-renderer';

test('checking jest/enzyme are working', () => {
  const num = 5;
  expect(2 + 3).toEqual(num);
})

describe('App', () => {
  let wrapper = shallow(<App />);

  it('shallow renders App', () => {
    wrapper;
  });

  it('Should render TaskComp', () => {
    expect(wrapper.find('TaskComp').length).toEqual(1);
  });
  
  it('Shouldn\'t render LogInComp', () => {
    expect(wrapper.find('LogInComp').length).toEqual(0);
  })

  // write test cases of what component should
  // render depending on the state values.
});


// snapshot testing

it('Renders App with TaskComp', () => {
  const AppComponent = renderer.create(<App />).toJSON();
  expect(AppComponent).toMatchSnapshot();
});






