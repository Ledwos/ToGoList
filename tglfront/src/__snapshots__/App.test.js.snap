// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App with TaskComp 1`] = `
<div>
  <p>
    I'm the task component
  </p>
  <p>
    the user id is 
    5
  </p>
  <div
    id="addTaskBtn"
    onClick={[Function]}
  >
    New Task
  </div>
  <div>
    <form
      method="POST"
      onSubmit={[Function]}
    >
      <label
        for="taskname"
      >
        Title:
      </label>
      <input
        id="taskname"
        required={true}
        type="text"
      />
      <label
        for="taskdesc"
      >
        Description:
      </label>
      <textarea
        id="taskdesc"
        placeholder="(optional)"
      />
      <label
        for="taskdate"
      >
        Date:
      </label>
      <div
        id="taskdate"
      >
        <label
          for="dateD"
        >
          DD:
        </label>
        <input
          id="dateD"
          max="31"
          min="1"
          placeholder="01"
          type="number"
        />
         -
        <label
          for="dateM"
        >
          MM:
        </label>
        <input
          id="dateM"
          max="12"
          min="1"
          placeholder="01"
          type="number"
        />
         -
        <label
          for="dateD"
        >
          YY:
        </label>
        <input
          id="dateY"
          max="2099"
          min={2020}
          placeholder={2020}
          type="number"
        />
      </div>
      <label
        for="tasktime"
      >
        Time:
      </label>
      <div
        id="tasktime"
      >
        <input
          id="timeH"
          max="23"
          min="0"
          placeholder="00"
          type="number"
        />
         :
        <input
          id="timeM"
          max="59"
          min="0"
          placeholder="00"
          type="number"
        />
      </div>
      <button
        type="submit"
      >
        Add!
      </button>
    </form>
  </div>
</div>
`;
